### PR TITLE
depends: Update intel-ipsec-mb to 1.2 for Win64

### PR DESCRIPTION
**Pull request by edtubbs**
Resolves [Win64 build with mingw comment in #2104](https://github.com/dogecoin/dogecoin/issues/2104#issuecomment-932993036)
Updated the version of intel-ipsec-mb in depends to 1.2
Added AVX2 experimental build with intel-ipsec-mb for Win64 to ci

### CODE REVIEW

**Comment by edtubbs** [comment]
### bench_dogecoin on Windows 10 (Intel i7)
```
c:\dogecoin>bench_dogecoin.exe
#Benchmark,count,min,max,average,min_cycles,max_cycles,average_cycles
Base58CheckEncode,327680,0.000003067776561,0.000003221139195,0.000003124927025,6461,6815,6596
Base58Decode,589824,0.000001593518391,0.000002025510184,0.000001768315593,3368,4293,3736
Base58Encode,589824,0.000001846678060,0.000002025510184,0.000001863247437,3887,4157,3936
CCheckQueueSpeed,12288,0.000076249940321,0.000106804072857,0.000084986968432,162568,221332,179612
CCheckQueueSpeedPrevectorJob,640,0.001555532217026,0.001723252236843,0.001661860942841,3252772,3639804,3509723
CCoinsCaching,131072,0.000007624941645,0.000008216069546,0.000007773620382,16088,17428,16412
CoinSelection,512,0.001865047961473,0.002317942678928,0.002004765905440,3931750,4931589,4231338
LockedPool,576,0.001799501478672,0.002012997865677,0.001904555906852,3853965,4255766,4021083
MempoolEviction,22528,0.000046700006351,0.000051467330195,0.000048139248975,99893,108747,101640
RIPEMD160,480,0.002165466547012,0.002196058630943,0.002171601851781,4555863,4682628,4587440
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000976000000000,0.000976000000000,0.000976000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000975000000000,0.000975000000000,0.000975000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom,1441792,0.000000670166628,0.000000895004632,0.000000748134963,1428,1893,1579
SHA1,640,0.001586005091667,0.001829877495766,0.001638817042112,3380698,3754764,3460438
SHA256,240,0.004087060689926,0.004453003406525,0.004200720787048,8730697,9425451,8875710
SHA256_32b,4,0.293056964874268,0.298655033111572,0.295855998992920,619359105,630705552,625032328
SHA512,416,0.002531498670578,0.002653531730175,0.002576079506140,5371205,5625888,5437685
Scrypt,4096,0.000243999995291,0.000261160079390,0.000251527817454,520964,553695,531229
SipHash_32b,44,0.022691011428833,0.025385499000549,0.023193316026167,48025353,53955424,49014905
Sleep100ms,10,0.100524544715881,0.101013541221619,0.100629091262817,212286944,213356119,212525537
Trig,25165824,0.000000039104179,0.000000042804913,0.000000040451842,82,90,85
```
### bench_dogecoin on Windows 10 (Intel i7 w/AVX2)
```
c:\dogecoin>avx2\bench_dogecoin.exe
#Benchmark,count,min,max,average,min_cycles,max_cycles,average_cycles
Base58CheckEncode,360448,0.000002889129973,0.000003335706424,0.000002973208294,6124,6854,6278
Base58Decode,655360,0.000001546897693,0.000001667969627,0.000001583751146,3296,3419,3345
Base58Encode,589824,0.000001787353540,0.000002144894097,0.000001848395186,3863,4422,3902
CCheckQueueSpeed,12288,0.000075294869021,0.000096264528111,0.000085542657568,159994,203075,180557
CCheckQueueSpeedPrevectorJob,640,0.001586005091667,0.001799467951059,0.001665300130844,3329343,3806755,3518150
CCoinsCaching,163840,0.000006254878826,0.000006859190762,0.000006455175753,13250,14538,13632
CoinSelection,512,0.001799494028091,0.002409778535366,0.002109240274876,3787358,5092325,4455352
LockedPool,512,0.001799188554287,0.002105358988047,0.001967332325876,3764839,4454128,4153503
MempoolEviction,20480,0.000040031736717,0.000062959035859,0.000049074948765,85064,133170,103675
RIPEMD160,480,0.002165496349335,0.002286404371262,0.002183802425861,4561239,4795624,4609883
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.001014000000000,0.001014000000000,0.001014000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000976000000000,0.000976000000000,0.000976000000000
RollingBloom-refresh,1,0.000977000000000,0.000977000000000,0.000977000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000976000000000,0.000976000000000,0.000976000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom,1441792,0.000000692498361,0.000000839057975,0.000000753880654,1465,1767,1591
SHA1,704,0.001402996480465,0.001532938331366,0.001454993066463,2987268,3231866,3073110
SHA256,352,0.002806000411510,0.003236033022404,0.002953329546885,5927839,6835216,6239353
SHA256_32b,6,0.225682020187378,0.268460988998413,0.248421986897786,475967689,567591831,524538290
SHA512,512,0.001814734190702,0.002104498445988,0.001972064375877,3833420,4466558,4165297
Scrypt,4096,0.000236367806792,0.000339312478900,0.000262650137302,503524,714192,554716
SipHash_32b,40,0.023180007934570,0.035623073577881,0.027417695522308,49282662,75614482,57910986
Sleep100ms,10,0.100511074066162,0.101521015167236,0.101017117500305,212181105,214385610,213311769
Trig,25165824,0.000000039092811,0.000000046314199,0.000000042507054,85,97,89

c:\dogecoin>
```

File: depends/packages/intel-ipsec-mb.mk [hunk 1]
```diff
@@ -1,19 +1,20 @@
 package=intel-ipsec-mb
-$(package)_version=1.0
+$(package)_version=1.2
 $(package)_download_path=https://github.com/intel/intel-ipsec-mb/archive/refs/tags
 $(package)_file_name=v$($(package)_version).tar.gz
-$(package)_sha256_hash=03501aea472d3c8fdf8f1f207816eefeaf5e4ebbdc71d88dcb26b2519841bb74
+$(package)_sha256_hash=f680b28369d02dc0978eb9b4bee6da9a132d66c666298e087a2b3e247548d99e
```
Comment (chromatic) [inline depends/packages/intel-ipsec-mb.mk:5]: I've downloaded this file and confirmed this checksum.

**Review by michilumin — APPROVED** [review]
Checked intel package checksum (matches), viewed code (OK) and built/ran bench. Did not do speed comparisons across different processors but looks OK to me to merge in terms of function.

**Comment by michilumin** [comment]
Which part? AFAIK this is going into experimental, doesn't change consensus, and @chromatic reviewed with an approval 19 days ago?

**Comment by patricklodder** [comment]
The part under "merging pull requests" where it says "we keep approved PRs open for 24h unless otherwise documented". This just had your approval (as did the other 2). I'll test from the branch instead, no big deal, but let's try to stick to the 24h so that more than one person has a chance to review - I just didn't get to it yet because I'm on vacay.

**Comment by michilumin** [comment]
Ok; I was under the impression here that they had chromatic's approval as well, and that they had hung open for well over 24h and were only stalled because @edtubbs forgot to remove the WIP flag.

**Comment by patricklodder** [comment]
No probs. Will propose some recommendations per PR.

**Pull request merged**